<a href="https://colab.research.google.com/github/602emerald/DSA-Bootcamp-Java/blob/main/Plastic_Emmission_to_Ocean_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import plotly.express as px

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

In [ ]:
path = '/content/plastics-top-rivers.csv'
df = pd.read_csv(path)

In [ ]:
df.info()               #Gives info on dataframe about Column Names, Object Type

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Entity         50 non-null     object 
 1   Country        50 non-null     object 
 2   sharePlastics  50 non-null     float64
dtypes: float64(1), object(2)
memory usage: 1.3+ KB


In [ ]:
df.shape

(50, 3)

In [ ]:
df.head(2)

Entity      Country  sharePlastics
0    Agno  Philippines       0.469647
1  Agusan  Philippines       0.469647

In [ ]:
df.rename(columns={"Entity":"River","sharePlastics":"Plastic_percent"},inplace=True)
df.columns

Index(['River', 'Country', 'Plastic_percent'], dtype='object')

In [ ]:
df.head()

River      Country  Plastic_percent
0            Agno  Philippines         0.469647
1          Agusan  Philippines         0.469647
2  Bharathappuzha        India         0.357340
3   Cagayan River  Philippines         0.234824
4  Cagayan de Oro  Philippines         0.265453

In [ ]:
df.tail()

River      Country  Plastic_percent
45         Ulhas        India         1.327265
46   Wouri River     Cameroon         0.285872
47  Yangon River      Myamnar         0.255243
48        Zapote  Philippines         0.275663
49      Zhujiang        China         0.255243

In [ ]:
df.isna().sum()    #checking for NaN values in dataframe

River              0
Country            0
Plastic_percent    0
dtype: int64

In [ ]:
df.describe()

Plastic_percent
count        50.000000
mean          0.539890
std           0.900586
min           0.224614
25%           0.255243
50%           0.290977
75%           0.426256
max           6.432129

In [ ]:
df.describe(include="all")    #describe function for non-numeric columns

River      Country  Plastic_percent
count     50           50        50.000000
unique    50           16              NaN
top     Agno  Philippines              NaN
freq       1           19              NaN
mean     NaN          NaN         0.539890
std      NaN          NaN         0.900586
min      NaN          NaN         0.224614
25%      NaN          NaN         0.255243
50%      NaN          NaN         0.290977
75%      NaN          NaN         0.426256
max      NaN          NaN         6.432129

In [ ]:
rivers = df['Country'].value_counts()
River_Total = pd.DataFrame(rivers)
River_Total.head(2)

count
Country           
Philippines     19
India            9

In [ ]:
River_Total.rename(columns={'Country':'river_total'},inplace=True)
River_Total

count
Country                  
Philippines            19
India                   9
Malaysia                5
China                   3
Cameroon                2
Myamnar                 2
Thailand                1
Indonesia               1
Cote d'Ivoire           1
Bangladesh              1
Sri Lanka               1
Nigeria                 1
Tanzania                1
Dominican Republic      1
Brazil                  1
Vietnam                 1

In [ ]:
River_Total['count'].sum()

50

In [ ]:
fig = px.bar(River_Total, x=River_Total.index, y="count")
fig.show()

In [ ]:
# Analysing River vs Plastic Share

In [ ]:
fig = px.bar(df,x='River',y='Plastic_percent',color='Country')
fig.show()

In [ ]:
fig = px.scatter(df,x='River',y='Plastic_percent',color='Country')
fig.show()

In [ ]:
fig = px.treemap(df, path=['Country','River'], color='Plastic_percent',)
fig.show()

In [ ]:
df['Plastic_percent'].sum()

26.994521456999998

In this notebook, we analyzed a dataset containing 50 rivers vs plastic shares information. Most of them belonging to the continent of Asia. We can conclude easily that only these 50 rivers contibute to approximately 26.99% of plastic deposition in the Oceans of our planets.

More than 1000 rivers account for 80% of global riverine plastic emmissions into the oceans.Source

This data may be old. But still this gives us an idea about the water pollution we humans cause. We all know that things are getting out of control.

WE HAVE ONE PLANET. IT'S OUR RESPONSIBILITY TO SAVE HER!